src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1282,8 +1315,9 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                 Utf8Output = utf8output,
                 CompilationName = compilationName,
                 OutputFileName = outputFileName,
+                OutputRefFilePath = outputRefFilePath,
                 PdbPath = pdbPath,
-                EmitPdb = emitPdb,
+                EmitPdb = emitPdb && !metadataOnly,
```

Comment:
>emitPdb && !metadataOnly [](start = 26, length = 24)

Code above checks the value of emitPdb. So if we are gonna change it here the check above won't take the metadataOnly flag into account.